indykite/resource_event_sink.go

```diff
@@ -134,6 +140,56 @@ func providerSchema() map[string]*schema.Schema {
 				},
 			},
 		},
+		azureEventGridKey: {
+			Type:        schema.TypeList,
+			MaxItems:    1,
+			Optional:    true,
```

Comment:
`azureEventGridKey`, `azureServiceBusKey` and `kafkaKey` all are optional. But still you can specify all of them.

I think you should add validation, that one of them must be specified. It should be possible I believe

Same for

Reply:
Ah... That sucks. Up to you. Either keep your solution if you have tests for it. Or just remove it

Reply:
validation kept